rotkehlchen/tests/unit/accounting/test_default_settings.py

```diff
@@ -0,0 +1,332 @@
+        is_taxable: bool,
+):
+    """Test that the default accounting settings for receiving are correct"""
+    _gain_one_ether(
+        transactions=accounting_pot.transactions,
+        event_type=event_type,
+        event_subtype=event_subtype,
+    )
+
+    expected_event = ProcessedAccountingEvent(
+        type=AccountingEventType.TRANSACTION_EVENT,
+        notes='Received 1 ETH',
+        location=Location.BLOCKCHAIN,
+        timestamp=TIMESTAMP_1_SECS,
+        asset=A_ETH,
+        free_amount=ZERO if is_taxable else ONE,
+        taxable_amount=ONE if is_taxable else ZERO,
+        price=Price(FVal(1862.06)),
+        pnl=PNL(
+            taxable=FVal(1862.06) if is_taxable else ZERO,
+            free=ZERO,
+        ),
+        cost_basis=None,
+        index=0,
+        extra_data={'tx_hash': EXAMPLE_TX_HASH_HEX},
+    )
+    expected_event.count_entire_amount_spend = False
+    expected_event.count_cost_basis_pnl = is_taxable
+
+    assert accounting_pot.processed_events == [expected_event]
+    assert accounting_pot.pnls.taxable == FVal(1862.06) if is_taxable else ZERO
+    assert accounting_pot.pnls.free == ZERO
+
+
+@pytest.mark.parametrize('event_type,event_subtype,is_taxable,counterparty,gas_taxable', [
+    (HistoryEventType.SPEND, HistoryEventSubType.NONE, True, None, False),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, True, None, False),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, True, CPT_GAS, True),
+    (HistoryEventType.SPEND, HistoryEventSubType.FEE, False, CPT_GAS, False),
+    (HistoryEventType.DEPOSIT, HistoryEventSubType.NONE, False, None, False),
+    (HistoryEventType.RENEW, HistoryEventSubType.NONE, True, None, False),
+])
+@pytest.mark.parametrize('mocked_price_queries', [prices])
+def test_accounting_spend_settings(
+        accounting_pot: 'AccountingPot',
+        event_type: 'HistoryEventType',
+        event_subtype: 'HistoryEventSubType',
+        is_taxable: bool,
+        counterparty: Optional[str],
+):
+    # Gain some ETH so that the spending event has something to spend
+    _gain_one_ether(transactions=accounting_pot.transactions)
+    spend_event = HistoryBaseEntry(
+        event_identifier=EXAMPLE_EVM_HASH,
+        sequence_index=0,
+        timestamp=TIMESTAMP_2_MS,
+        location=Location.BLOCKCHAIN,
+        location_label=EXAMPLE_ADDRESS,
+        asset=A_ETH,
+        balance=Balance(amount=FVal(0.5)),
+        notes='Send 1 ETH to 0xABC',
+        event_type=event_type,
+        event_subtype=event_subtype,
+        counterparty=counterparty,
+    )
+    consumed_num = accounting_pot.transactions.process(event=spend_event, events_iterator=iter([]))
+    assert consumed_num == 1
+
+    acquisition_event = AssetAcquisitionEvent(
+        amount=ONE,
+        timestamp=TIMESTAMP_1_SECS,
+        rate=Price(FVal(1862.06)),
+        index=0,
+    )
+    acquisition_event.remaining_amount = FVal(0.5)
+    cost_basis = None
+    if is_taxable:
```

Comment:
So some comments here:

1. Add comments, with the calculations and what it is you are testing.
2. You are mocking stuff. Use easier numbers. Why the heck use `1862.06` and `3258.55`. You know you can **and should** mock price queries. We have fixtures for this. Make sure no network calls happen in all these tests you made.
3. The spending part and the selling part are different and should be checked for in the test depending on customization. This is what `count_entire_amount` as spend is for and `taxable + crypto_to_crypto`.

Reply:
Used already existing big structure with mocked prices and because of that prices were so ugly. But now improved.